modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFormProviderTest.java

```diff
@@ -0,0 +1,155 @@
+
+import java.util.Arrays;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Jürgen Kappler
+ */
+@RunWith(Arquillian.class)
+public class ObjectEntryInfoItemFormProviderTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-176083", "true"
+			).build());
+
+		_objectDefinition =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				TestPropsValues.getUserId(), false,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"A" + RandomTestUtil.randomString(), null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectDefinitionConstants.SCOPE_SITE,
+				ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
+				Arrays.asList(_getBusinessTypeAttachmentObjectField()));
+
+		_objectDefinition =
+			_objectDefinitionLocalService.publishCustomObjectDefinition(
+				TestPropsValues.getUserId(),
+				_objectDefinition.getObjectDefinitionId());
+	}
+
+	@After
+	public void tearDown() {
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-176083", "false"
+			).build());
+	}
+
+	@Test
+	public void testObjectEntryInfoItemFormProviderWithAttachment()
+		throws Exception {
+
+		InfoItemFormProvider<?> infoItemFormProvider =
+			_infoItemServiceRegistry.getFirstInfoItemService(
+				InfoItemFormProvider.class, _objectDefinition.getClassName());
+
+		InfoForm infoForm = infoItemFormProvider.getInfoForm(
+			String.valueOf(_objectDefinition.getObjectDefinitionId()));
+
+		Assert.assertNotNull(infoForm);
+
+		InfoField<?> attachmentInfoField = infoForm.getInfoField("attachment");
+
+		Assert.assertNotNull(attachmentInfoField);
+	}
+
+	private ObjectFieldSetting _createObjectFieldSetting(
+		String name, String value) {
+
+		ObjectFieldSetting objectFieldSetting =
+			_objectFieldSettingLocalService.createObjectFieldSetting(0L);
+
+		objectFieldSetting.setName(name);
+		objectFieldSetting.setValue(value);
+
+		return objectFieldSetting;
+	}
+
+	private ObjectField _getBusinessTypeAttachmentObjectField() {
+		ObjectFieldBuilder objectFieldBuilder = new ObjectFieldBuilder();
+
+		return objectFieldBuilder.businessType(
+			ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT
+		).dbType(
+			ObjectFieldConstants.DB_TYPE_LONG
+		).indexed(
+			true
+		).labelMap(
+			LocalizedMapUtil.getLocalizedMap("Attachment")
+		).name(
+			"attachment"
+		).objectFieldSettings(
+			Arrays.asList(
+				_createObjectFieldSetting("acceptedFileExtensions", "txt"),
+				_createObjectFieldSetting("fileSource", "userComputer"),
+				_createObjectFieldSetting("maximumFileSize", "100"))
+		).build();
+	}
```

Comment:
@jkappler 

replace `ObjectFieldBuilder` with `AttachmentObjectFieldBuilder`:

````java
private ObjectField _getAttachmentObjectField() {
    return new AttachmentObjectFieldBuilder().indexed(
	    true
    ).labelMap(
	    LocalizedMapUtil.getLocalizedMap("Attachment")
    ).name(
	    "attachment"
    ).objectFieldSettings(
	    Arrays.asList(
		    _createObjectFieldSetting("acceptedFileExtensions", "txt"),
		    _createObjectFieldSetting("fileSource", "userComputer"),
		    _createObjectFieldSetting("maximumFileSize", "100"))
    ).build();
}
````

rename to `_getAttachmentObjectField`.

Thank you!

Reply:
Thanks for letting me know, I didn't know it existed!